Compute Identity fails when `mtime` is missing in a message

I think this is a bug. `mtime` is stated to be an optional parameter in the documentation : https://metpx.github.io/sarracenia/Reference/sr_post.7.html#:~:text=%22mtime%22%20%3A%20date%20string%20%2D%20last%20modification%20time%20of%20a%20file%20(optional).

When computeIdentity is called and `mtime` is missing in a message, the file gets retried.

```
2026-03-06 05:55:37,169 [INFO] convert.imv6_to_png download converting file.imv6 to file.png
2026-03-06 05:55:37,445 [INFO] convert.imv6_to_png download Exception details:
Traceback (most recent call last):
  File "/local/home/sarra/.config/sr3/plugins/convert/imv6_to_png.py", line 121, in download
    msg.computeIdentity(opath, self.o)
  File "/usr/lib/python3/dist-packages/sarracenia/__init__.py", line 542, in computeIdentity
    if xattr.get('mtime') >= msg['mtime']:
KeyError: 'mtime'
...
2026-03-06 05:55:38,033 [ERROR] sarracenia.flow do_download gave up downloading for now, appending to retry queu
```

I think this should be a pretty easy fix in the `computeIdentity` code.

--- a/sarracenia/__init__.py
+++ b/sarracenia/__init__.py
@@ -538,7 +538,7 @@ class Message(dict):
                     'random', 'md5', 'md5name', 'sha512', 'cod,md5', 'cod,sha512'
                 ]
                 calc_method = random.choice(methods)
-            elif 'identity' in xattr.x and 'mtime' in xattr.x:
+            elif 'identity' in xattr.x and 'mtime' in xattr.x and 'mtime' in msg:
                 if xattr.get('mtime') >= msg['mtime']:
                     logger.debug("mtime remembered by xattr")
                     fxainteg = xattr.get('identity')

